command/flags.go

```diff
@@ -204,6 +204,12 @@ var initFlags = []cli.Flag{
 		EnvVars:  []string{"STORETHEINDE_PUBSUB_TOPIC"},
 		Required: false,
 	},
+	&cli.BoolFlag{
+		Name:     "upgrade",
+		Usage:    "Upgrade the config file to the current version, saving the old config as config.prev, and ignoring other flags ",
```

Comment:
when does this need to be run? (how do we specify getting it run in a production environment, is it safe to just always have this set?)

Reply:
This does not need to be run, as the config will already ignore obsolete values and set the defaults for unconfigured values.  However, this should be run when a user wants to upgrade their config file without having to run `init` and then copy all the old values that are still used from the old config to the new.

I will output a message to stdout when running the daemon, to tell the user they should upgrade their configuration.

I would like to have this run automatically on detecting a new version, but we track config changes in github so it would not be good to have something change the file automatically.  Therefore, I made it something that must be run manually.

Reply:
we should figure out a path that makes the deployment story easier, like a default '--upgrade-if-new-version' style thing so we don't need manually changing the deploy flags on each time and then back

Reply:
A `--upgrade-if-new-version` would change the config on the host without updating it in [the github repo](https://github.com/filecoin-project/lotus-infra/blob/master/kubernetes/mainnet-us-east-1-eks/helm/ntwk-mainnet-storetheindex/filecoin/lotus-bundle/storetheindex-0.yaml#L41-L102) wouldn't it?.

If we need to check a new config file into github to update the config, then it seems that someone must upgrade manually and then checks the new config into github. That then gets picked up by the next deploy, and no deploy flags are changed (this is how it would work with our current prod).